src/http/Response.cpp

```diff
@@ -0,0 +1,208 @@
+}
+
+void
+Response::load_response_map(int status_code)
+{
+	_response_map["Date"] += get_time_stamp();
+	_response_map["Status-line"]
+		= _response_map["Protocol"] + _status_code.get_key_value_formated(status_code);
+	if (status_code != 200)
+	{
+		set_response_type(".html");
+		create_error_html_page(status_code);
+	}
+	else
+	{
+		set_response_type(_response_map["dir_location"]);
+		construct_body_string(_response_map["dir_location"]);
+	}
+	set_content_length(_response_map["body-string"]);
+	construct_header_string();
+	construct_full_response();
+}
+
+std::string
+Response::get_time_stamp(void)
+{
+	std::time_t stamp = std::time(NULL);
+	std::string formated_date = std::asctime(std::gmtime(&stamp));
+	formated_date = formated_date.substr(0, formated_date.length() - 1);
+	formated_date += " GMT";
+	return formated_date;
+}
+
+void
+Response::set_content_length(std::string str)
+{
+	_response_map["Content-Length"] = std98::to_string(str.length());
+}
+
+bool
+Response::file_exists(std::string path)
+{
+	std::ifstream file;
+ 	bool		  ret = false;
+	file.open(path.c_str());
+	if (file)
+	{
+		ret = true;
+		file.close();
+	}
+	return ret;
+}
+
+void
+Response::set_response_type(std::string path)
+{
+	std::string type = "";
+	type = path.substr(path.rfind(".") + 1, path.size() - path.rfind("."));
+	if (type == "html")
+		_response_map["Content-Type"] = "text/html";
+	else if (type == "css")
+		_response_map["Content-Type"] = "text/css";
+	else if (type == "js")
+		_response_map["Content-Type"] = "text/javascript";
+	else if (type == "jpeg" || type == "jpg")
+		_response_map["Content-Type"] = "image/jpeg";
+	else if (type == "png")
+		_response_map["Content-Type"] = "image/png";
+	else if (type == "bmp")
+		_response_map["Content-Type"] = "image/bmp";
+	else if (type == "ico")
+		_response_map["Content-Type"] = "image/x-icon";
+	else
+		_response_map["Content-Type"] = "text/plain";
+}
+
+void
+Response::construct_header_string(void)
+{
+	std::string CRLF = "\r\n";
+
+	_response_map["header-string"] += _response_map["Status-line"];
+	_response_map["header-string"] += CRLF;
+	_response_map["header-string"] += "Date: " + _response_map["Date"];
+	_response_map["header-string"] += CRLF;
+	_response_map["header-string"] += "Server: " + _response_map["Server"];
+	_response_map["header-string"] += CRLF;
+	_response_map["header-string"] += "Content-Length: " + _response_map["Content-Length"];
+	_response_map["header-string"] += CRLF;
+	_response_map["header-string"] += "Content-Type: " + _response_map["Content-Type"];
+	_response_map["header-string"] += CRLF;
+	_response_map["header-string"] += "Connection: " + _response_map["Connection"];
+	_response_map["header-string"] += CRLF;
+	_response_map["header-string"] += CRLF;
+}
+
+void
+Response::construct_body_string(std::string path_to_file)
+{
+	std::ifstream	  file;
+	std::stringstream buffer;
+
+	file.open(path_to_file.c_str());
+	if (file.fail())
+	{
+		std::cout << "Open file error" << std::endl;
```

Comment:
Write the error in the STDERR instead of the STDOUT ?
```suggestion
		std::cerr << "Open file error" << std::endl;
```

Reply:
Accepted